File: microservices/find/models/Option.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;


const optionSchema = Schema({
    iso: { type: String, required: true },
    sort: { type: Number, required: false, default: 0 },
    name: { type: String, required: true, unique: true },
    slug: { type: String, required: false, unique: true },
    description: { type: String, required: false, default: '' },
    media: { type: String, default: '' },
    unit: { type: String, required: false, default: '' },
    maximum: { type: Number, default: 0 },
    minimum: { type: Number, default: 0 },
    incremental_by: { type: Number, default: 0 },
    information: { type: String, default: '' },
    published: { type: Boolean, default: true },
    input_type: { type: String, default: '' },
    created_at: { type: Date, required: true },
    // children = db.ListField(db.ReferenceField('self', default=None),  default=None)
});


module.exports = Option = mongoose.model("options", optionSchema);

File: plugins/probo/models/option.py
from database.db import db
from flask_mongoengine import BaseQuerySet
import datetime
from slugify import slugify, Slugify, UniqueSlugify
from models.box import Box
from flask_mongoengine.wtf import model_form


class Option(db.Document):
    iso = db.StringField(required=True)
    sort = db.IntField(required=False, default=0)
    name = db.StringField(required=True, unique=True)
    slug = db.StringField(required=False, unique=True)
    description = db.StringField(required=False, default='')
    media = db.StringField(default='')
    incremental_by = db.IntField(min_value=None, max_value=None, default=0)
    information = db.StringField(default='')
    box = db.StringField(default='')
    published = db.BooleanField(default=True)
    input_type = db.StringField(default='')
    created_at = db.DateTimeField(default=datetime.datetime.now, required=True)
    valid_options = db.ListField(default=None)

    meta = {
        'indexes': ['-created_at', 'slug', 'name'],
        'ordering': ['-created_at', 'name'],
        'collection': 'dwd_options',
        'queryset_class': BaseQuerySet
    }

    def __setattr__(self, key, value):
        super(Option, self).__setattr__(key, value)
        if key == 'name':
            self.slug = slugify(self.name, to_lower=True)
